source/Projectile.cpp

```diff
@@ -159,7 +159,13 @@ void Projectile::Move(vector<Visual> &visuals, vector<Projectile> &projectiles)
 	double accel = weapon->Acceleration();
 	int homing = weapon->Homing();
 	if(target && homing && !Random::Int(30))
+	{
 		CheckLock(*target);
+		CheckConfused(*target);
+	}
+	// Make sure to set a confusion direction if there isn't one.
```

Comment:
```suggestion
	// Make sure to set a confusion direction if there isn't one.
	// Also update the direction after the missile turns about 90 degrees away from its target.
```

`confusionDirection` is only 0 initially, right? We don't need to check that every frame; we could just assign an initial value when the projectile is constructed.